Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -184,6 +184,29 @@ int HardwareBreakpointManager::getAvailableLocation() {
 
   return (it - _locations.begin());
 }
+
+ErrorCode HardwareBreakpointManager::isValid(Address const &address,
+                                             size_t size, Mode mode) const {
+  switch (size) {
+  case 1:
+    break;
+  case 8:
+    DS2LOG(Warning, "8-byte breakpoints not supported on all architectures");
```

Comment:
Oh yes, that. :laughing:

On Wed, Jun 8, 2016 at 6:27 PM, Francis Ricci <notifications@github.com>
wrote:

> In Sources/Architecture/X86/HardwareBreakpointManager.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r66369863>:
>
> > @@ -184,6 +184,29 @@ int HardwareBreakpointManager::getAvailableLocation() {
> >
> >    return (it - _locations.begin());
> >  }
> > +
> > +ErrorCode HardwareBreakpointManager::isValid(Address const &address,
> > +                                             size_t size, Mode mode) const {
> > +  switch (size) {
> > +  case 1:
> > +    break;
> > +  case 8:
> > +    DS2LOG(Warning, "8-byte breakpoints not supported on all architectures");
>
> We talked about this before haha, it's more complicated than just x86 vs
> x86_64:
>
> For Pentium® 4 and Intel® Xeon® processors with a CPUID signature
> corresponding to family 15 (model 3, 4, and 6), break point conditions
> permit specifying 8-byte length on data read/write with an of encoding 10B
> in the LENn field.
> Encoding 10B is also supported in processors based on Intel Core
> microarchitecture or enhanced Intel Core microarchitecture, the respective
> CPUID signatures corresponding to family 6, model 15, and family 6,
> DisplayModel value 23 (see CPUID instruction in Chapter 3, “Instruction Set
> Reference, A-M” in the Intel® 64 and IA-32 Architectures Software
> Developer’s Manual, Volume 2A). The Encoding 10B is supported in processors
> based on Intel® AtomTM microarchitecture, with CPUID signature of family 6,
> DisplayModel value 1CH. The encoding 10B is undefined for other processors.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/f1abcb6603459d8af3c526c2c1941f5474ea14d6..bc75320b63d4c9ed10847d4972ed3d4b4933eeba#r66369863>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADHbC52Z58bVrIZJI4jWpn-0oSodRYpks5qJ2wegaJpZM4H9FRy>
> .
>



-- 
Stephane Sezer